woocommerce/woocommerce-admin
# Hook up download button to export endpoint.

In support of #311. 

Also moves comparison, search, and download actions to ReportTable component.

**_NOTE:_**

This PR now contains the contents of https://github.com/woocommerce/woocommerce-admin/pull/2951 and https://github.com/woocommerce/woocommerce-admin/pull/2899 as well. It will serve as the roll-up PR for all CSV download UI changes.

This PR seeks to move the `<Card>` actions from the published `TableCard` component to the `ReportTable` component in WooCommerce Admin where the actions are used.

The rationale here being:
- There was too much of the WooCommerce Admin specific report UX in the `TableCard` component (it even used the `woocommerce-analytics__card` class, which would probably break for consumers)
- We will likely need to make changes to the download behavior, which was solely in the `TableCard` component

In addition to moving the compare, search, and download actions to `ReportTable`, this PR also moves the IE-specific styles for those items which is a new feature for WooCommerce Admin. A new Webpack entry `ie` has been created so that a `dist/ie/style.css` is created. The `FixStyleOnlyEntriesPlugin` plugin prunes the unneeded `dist/ie/index.js` that Webpack is compelled to create.

### Detailed test instructions:

- Check all Analytics tables, verify they function as before
- Open a report with more only 1 page of results
- Click download
- Verify a .csv file is immediately downloaded
- Open a report with more only 1 page of results
- Click download
- Verify a notice is displayed stating you'll receive an email
- Verify you receive an email, and the download link works



### Changelog Note:



Tweak: Moved WC-Admin specific actions from TableCard to ReportTable.
